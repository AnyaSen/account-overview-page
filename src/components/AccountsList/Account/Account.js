import React from "react";

import Styles from "./Account.module.scss";

export default function Account({ name, ID, balance, active }) {
  return (
    <div data-testid="Account" className={Styles.Account}>
      <div
        data-testid="active-pin"
        className={active ? Styles.activePin : Styles.inactivePin}
      ></div>

      <p data-testid="AccountName">Name: {name}</p>
      <p data-testid="AccountID">ID: {ID}</p>
      <p data-testid="AccountBalance">Balance: {balance}</p>
    </div>
  );
}
