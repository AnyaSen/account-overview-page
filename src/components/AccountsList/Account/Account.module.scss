@import "../../../assets/scss/variables";
@import "../../../assets/scss/breakpoints";

.Account {
  margin: $space-2;
  padding: $space-1;
  width: fit-content;
  box-shadow: $shadow-1;
  position: relative;
  height: fit-content;

  @include sm {
    margin: $space-2 $space-1;
    font-size: 5rem;
  }

  .activePin {
    background-color: $color-green;
    height: 1rem;
    width: 1rem;
    border-radius: 0.5rem;
    margin-top: -$space-2;
    margin-left: -$space-2;
  }

  .inactivePin {
    @extend .activePin;

    background-color: $color-red;
  }
}
