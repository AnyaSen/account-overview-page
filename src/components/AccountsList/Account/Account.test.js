import React from "react";
import { render, cleanup } from "@testing-library/react";

import Account from "./Account";

describe("<Account />", () => {
  afterAll(cleanup);

  test("Should have correct name, id, balance and active status.", () => {
    const name = "FUNDING_ACCOUNT";
    const id = "ZCPN537";
    const balance = 60;
    const active = true;

    const { getByText, getByTestId, container } = render(
      <Account name={name} ID={id} balance={balance} active={active} />
    );

    const nameText = getByText(`Name: ${name}`);
    const IDText = getByText(`ID: ${id}`);
    const balanceValue = getByText(`Balance: ${balance}`);
    const activeStatus = getByTestId("active-pin");

    expect(nameText).toBeTruthy();
    expect(IDText).toBeTruthy();
    expect(balanceValue).toBeTruthy();
    expect(activeStatus).toHaveClass("activePin");

    expect(container.firstChild).toMatchSnapshot();
  });

  test("Should have inactive status.", () => {
    const active = false;

    const { getByTestId } = render(<Account active={active} />);

    const activeStatus = getByTestId("active-pin");

    expect(activeStatus).toHaveClass("inactivePin");
  });
});
