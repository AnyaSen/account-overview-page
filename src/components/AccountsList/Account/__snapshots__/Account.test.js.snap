// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Account /> Should have correct name, id, balance and active status. 1`] = `
<div
  class="Account"
  data-testid="Account"
>
  <div
    class="activePin"
    data-testid="active-pin"
  />
  <p
    data-testid="AccountName"
  >
    Name: 
    FUNDING_ACCOUNT
  </p>
  <p
    data-testid="AccountID"
  >
    ID: 
    ZCPN537
  </p>
  <p
    data-testid="AccountBalance"
  >
    Balance: 
    60
  </p>
</div>
`;
